src/librustc_metadata/diagnostics.rs

```diff
@@ -26,6 +26,16 @@ name. Example:
 ```
 "##,
 
+E0455: r##"
+A framework kind link was used and the OS utilized to compile isn't the OSX. Erroneous code example:
+
+```
+#[link(name = "FooCoreServices",  kind = "framework")] extern {} // OS used to compile is Linux for example
+```
+
+When using link kind of 'framework' ensure to utilize OSX to compile the code.
```

Comment:
This sentence sounds strange for me. Did you mean "When using 'framework' link kind..."?